### PR TITLE
HYPERFLEET-864 - fix: support Go template lists and conditionals in manifest resources

### PR DESCRIPTION
**Pull request by xueli181114**
## Summary
- Manifest `ref:` files are now loaded as raw strings instead of being parsed as YAML, allowing structural Go templates (`{{ if }}`, `{{ range }}`, `{{ else }}`) to be used for conditional properties and list generation
- Inline manifests support the same templates via YAML multiline string syntax (`manifest: |`)
- Templates are rendered at execution time (when params are available), then the rendered output is parsed as YAML
- K8s manifest validation is skipped for string manifests since validation cannot occur before template rendering

### Changes
- **loader.go**: New `loadRawFile()` function; `loadTaskConfigFileReferences` stores manifest refs as raw strings
- **resource_executor.go**: New `renderStringManifest()` method handles string manifests (render template → parse YAML → marshal JSON)
- **validator.go**: Skip K8s manifest validation for string manifests (deferred to execution time)

## Test plan
- [x] Existing unit tests updated and passing
- [x] New tests for structural Go templates (`{{ if }}`, `{{ range }}`, `{{ if/else }}`)
- [x] New test for customer's subnet list use case
- [x] New tests for ref file and inline string manifest loading
- [ ] CI pipeline passes

Relates to: [HYPERFLEET-864](https://redhat.atlassian.net/browse/HYPERFLEET-864)

## Summary by CodeRabbit

* **New Features**
  * Referenced or inlined manifests are now stored as raw template strings and rendered at runtime; GVK info can be extracted from template-containing manifests to enable execution without prior YAML parsing.
  * Build script now detects container platform for image builds.

* **Bug Fixes**
  * Validation defers schema checks for template-based manifests to avoid false errors.

* **Tests**
  * Broader test coverage for templated manifests, GVK resolution, rendering edge cases, and end-to-end flows.

* **Chores**
  * Integration image base updated for CI.

### CODE REVIEW

File: internal/configloader/validator_test.go [hunk 1]
```diff
@@ -349,6 +349,54 @@ func TestValidateK8sManifests(t *testing.T) {
...
+		cfg := baseTaskConfig()
+		cfg.Resources = []Resource{{
+			Name:      "testResource",
+			Manifest:  `{{ if .condition }}full: template{{ end }}`,
+			Discovery: &DiscoveryConfig{Namespace: "*", ByName: "test"},
+		}}
+		v := newTaskValidator(cfg)
+		require.NoError(t, v.ValidateStructure())
+		require.NoError(t, v.ValidateSemantic())
+	})
```
[the remark this answers is not in the document]
Comment (xueli181114) [reply internal/configloader/validator_test.go]: Skipping. The design intentionally defers all K8s validation for string manifests to execution time. Adding template-delimiter detection (`{{` scanning) would add complexity for minimal gain — non-templated ref files are rare, and the error surfaces clearly at execution time regardless.

File: test/Dockerfile.integration [hunk 2]
```diff
@@ -2,17 +2,18 @@
...
 # This eliminates the need for dynamic installation during test runtime
 
-FROM golang:1.25-alpine
+FROM registry.access.redhat.com/ubi9/go-toolset:1.25
 
-# Install runtime dependencies
-RUN apk add --no-cache curl tar openssl bash
+# Install runtime dependencies as root, then switch back to non-root.
+USER root
+RUN dnf install -y tar && dnf clean all
```
[the remark this answers is not in the document]
Comment (xueli181114) [reply test/Dockerfile.integration]: Fixed in f9fe6cc. Added `USER 1001` after privileged operations and `chown` on `/tmp/envtest` directories.

File: internal/manifest/gvk.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// ExtractGVKFromString extracts apiVersion and kind from a YAML string
+// by scanning lines. This handles manifests with Go template directives
+// that would fail full YAML parsing.
+func ExtractGVKFromString(manifest string) schema.GroupVersionKind {
```
Comment (rh-amarin) [inline internal/manifest/gvk.go:17]: Do we need to render the template before searching for the GVK?

Silly example, but this will fail
```
apiVersion: work.open-cluster-management.io/v1
{{if eq .adapterName "dry-run-adapter"}}
kind: ManifestWork{{ .adapterName}}-1
{{else}}
kind: ManifestWork{{ .adapterName}}-2
{{end}}
```

But imagine having the kind being dynamic or the version

Comment (xueli181114) [reply internal/manifest/gvk.go:17]: `apiVersion` and `kind` are always static K8s resource types — the API server requires exact schema identifiers, so they cannot be dynamic. The line-scanning approach relies on this assumption and works because it only extracts those two inherently static fields.

Additionally, in the deletion flow (discover → delete, no manifest rendering), there is no rendered object to extract GVK from — so raw template scanning is the only option. This further validates the static-GVK design.

Will add a comment to document this constraint.

File: internal/manifest/render.go [hunk 4]
```diff
@@ -2,60 +2,66 @@
...
 		}
-
-		renderedValue, err := renderValue(v, params)
+		return string(data), nil
+	case map[interface{}]interface{}:
+		converted := utils.ConvertToStringKeyMap(m)
+		data, err := yaml.Marshal(converted)
 		if err != nil {
-			return nil, fmt.Errorf("failed to render value for key '%s': %w", k, err)
+			return "", fmt.Errorf("failed to marshal manifest to YAML: %w", err)
```
Comment (rh-amarin) [inline internal/manifest/render.go:29]: Do these two cases make sense?
Aren't we now treating manifests always as go templates? meaning that they will be always strings?

Comment (xueli181114) [reply internal/manifest/render.go:29]: Good question. Yes, we now always treat manifests as Go templates at **render time**. But the config loader still produces different types depending on how the manifest is authored:

- **Inline manifest** (plain YAML under `manifest:`) → YAML decoder parses it into `map[string]interface{}` before our code runs. We don't control this — it's the YAML library behavior.
- **Block scalar** (`manifest: |`) or **file ref** (`manifest.ref:`) → arrives as a raw `string`.

`ToYAMLString` bridges that gap by converting maps back to strings so the unified Go template path can take over. The config loader itself wasn't changed in this PR.

Moving the string conversion upstream into the config loader (so manifests are always stored as strings from the start) would be a cleaner design but a larger refactor — could be a follow-up.
